cms/settings/base.py

```diff
@@ -179,12 +180,45 @@
 
 
 # Database
-# This setting will use DATABASE_URL environment variable.
-# https://docs.djangoproject.com/en/stable/ref/settings/#databases
-# https://github.com/kennethreitz/dj-database-url
 
-DATABASES = {"default": dj_database_url.config(conn_max_age=600, default="postgres:///ons")}
+db_conn_max_age = int(env.get("PG_CONN_MAX_AGE", 870))  # Just under 15 minutes, to match password expiry
 
+if "PG_DB_ADDR" in env:
+    # Use IAM authentication to connect to the Database
+    DATABASES = {
+        "default": cast(
+            dj_database_url.DBConfig,
+            {
+                "ENGINE": "django_iam_dbauth.aws.postgresql",
+                "NAME": env["PG_DB_DATABASE"],
+                "USER": env["PG_DB_USER"],
+                "HOST": env["PG_DB_ADDR"],
+                "PORT": env["PG_DB_PORT"],
+                "CONN_MAX_AGE": db_conn_max_age,
+                "OPTIONS": {"use_iam_auth": True, "sslmode": "require"},
+            },
+        )
+    }
+
+    # Additionally configure a read-replica
+    if "PG_DB_READ_ADDR" in env:
+        DATABASES["read_replica"] = {**deepcopy(DATABASES["default"]), "HOST": env["PG_DB_READ_ADDR"]}
+
+else:
+    DATABASES = {
+        "default": dj_database_url.config(conn_max_age=db_conn_max_age, default="postgres:///ons"),
+    }
+
+    if "READ_REPLICA_DATABASE_URL" in env:
+        DATABASES["read_replica"] = dj_database_url.config(
+            env="READ_REPLICA_DATABASE_URL", conn_max_age=db_conn_max_age
+        )
+
+if "read_replica" not in DATABASES:
+    DATABASES["read_replica"] = deepcopy(DATABASES["default"])
```

Comment:
We don't want a separate DB for the read replica, because reads need to go to the same database as writes.

I don't think there'll be any downsides, besides more database connection handling. Isolated connections to the same DB should be functionally the same as the production read replica setup anyway.

Reply:
Sounds good to me. I think what would help once the infra is set up is to have a page in the docs outlining the read/write replicas.